draft-ietf-quic-http.md

```diff
@@ -401,6 +538,17 @@ All frames have the following format:
 
 ## Frame Definitions {#frames}
 
+### HAS_BODY {#frame-has-body}
+
+The HAS_BODY frame (type=0x0) indicates that the HTTP message contains a payload
+body.  It is sent after the initial header block on a request, response, or push
+stream.
+
+The HAS_BODY frame has no flags and is always empty.  A HAS_BODY frame with a
+non-empty payload or flags that are set MUST be treated as a
+HTTP_MALFORMED_HAS_BODY error.
```

Comment:
I considered that, and it's definitely an optimization that we can talk about later, so I went with the simplest option I could think of. The problem with the header block is that there are potentially multiple places you would put this.  The easiest approach is to grab a flag, but those appear in multiple places, so does this have to be consistent, or do you only allowing if END_HEADERS is also set, or can it appear on any HEADERS frame?